src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs

```diff
@@ -1904,7 +1906,7 @@ internal void Remove(bool notify)
                 PInvoke.SendMessage(tv, PInvoke.TVM_DELETEITEM, 0, (LPARAM)HTREEITEMInternal);
             }
 
-            _treeView._nodesByHandle.Remove(HTREEITEMInternal);
+            _treeView!._nodesByHandle.Remove(HTREEITEMInternal);
```

Comment:
Consider changing to
```suggestion
            tv._nodesByHandle.Remove(HTREEITEMInternal);
```